freqtrade/exchange/exchange.py

```diff
@@ -2000,10 +2077,39 @@ def _build_coroutine(
             else:
                 # Time jump detected, evict cache
                 logger.info(
-                    f"Time jump detected. Evicting cache for {pair}, {timeframe}, {candle_type}")
+                    f"Time jump detected. Evicting ohlcv cache for {pair}, {timeframe}, {candle_type}")
                 del self._klines[(pair, timeframe, candle_type)]
 
         if (not since_ms and (self._ft_has["ohlcv_require_since"] or not_all_data)):
+            # Multiple calls for one pair - to get more history
+            since_ms = self.needed_candle_ms(timeframe, candle_type)
```

Comment:
I extracted this lines into `needed_candle_ms` because it's needed in both variants of `build_coroutine`. Should I revert this change as well (and then create a PR) or is this ok?

Reply:
reducing code duplication will be fine :+1: